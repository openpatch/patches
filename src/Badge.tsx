/** @jsx jsx */
import { ReactNode } from "react";
import { jsx, css, Theme } from "@emotion/react";
import styled from "@emotion/styled";
import { Box } from "./Box";
import { variant } from "./system";
import { SystemColor, SystemColorNames } from "./system/types";

export type BadgeProps = {
  children?: ReactNode;
  tone?: SystemColorNames;
  icon?: ReactNode;
};

const StyledBadge = styled(Box)(
  variant<
    {
      color: SystemColor;
      backgroundColor: SystemColor;
      borderColor: SystemColor;
    },
    SystemColorNames,
    string
  >({
    scale: "badges",
    prop: "tone",
    variants: {
      primary: {
        color: "primary.800",
        backgroundColor: "primary.100",
        borderColor: "primary.800",
      },
      accent: {
        color: "accent.800",
        backgroundColor: "accent.100",
        borderColor: "accent.800",
      },
      neutral: {
        color: "neutral.800",
        backgroundColor: "neutral.100",
        borderColor: "neutral.800",
      },
      success: {
        color: "success.800",
        backgroundColor: "success.100",
        borderColor: "success.800",
      },
      warning: {
        color: "warning.800",
        backgroundColor: "warning.100",
        borderColor: "warning.800",
      },
      error: {
        color: "error.800",
        backgroundColor: "error.100",
        borderColor: "error.800",
      },
      info: {
        color: "info.800",
        backgroundColor: "info.100",
        borderColor: "info.800",
      },
    },
  })
);

export const Badge = ({ icon, tone = "primary", children }: BadgeProps) => {
  return (
    <StyledBadge
      tone={tone}
      display="inline-flex"
      alignItems="center"
      justifyContent="center"
      fontSize="xsmall"
      fontWeight="semibold"
      px="xsmall"
      py="xxsmall"
      borderRadius="full"
      borderStyle="solid"
      borderWidth="standard"
    >
      {icon && (
        <Box
          as="span"
          display="inline-flex"
          alignItems="center"
          justifyContent="center"
          width="12px"
          mr="xxsmall"
        >
          {icon}
        </Box>
      )}
      {children}
    </StyledBadge>
  );
};
