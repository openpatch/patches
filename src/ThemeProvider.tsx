import React, { ReactElement } from "react";
import { ThemeProvider as EThemeProvider } from "emotion-theming";
import { Theme, base } from "./themes";
import { GlobalCssTheme } from "./themes/utils";

interface ThemeProviderProps {
  theme?: Theme;
  children: ReactElement;
}

const ThemeProvider = ({ theme = base, children }: ThemeProviderProps) => {
  return (
    <EThemeProvider theme={theme}>
      <GlobalCssTheme theme={theme} />
      {children}
    </EThemeProvider>
  );
};

export default ThemeProvider;
