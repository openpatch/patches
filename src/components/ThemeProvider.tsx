import React, { ReactElement } from "react";
import {
  ThemeProvider as SCThemeProvider,
  DefaultTheme,
} from "styled-components";
import defaultTheme from "../shared/theme";

interface ThemeProviderProps {
  theme?: DefaultTheme;
  children: ReactElement;
}

const ThemeProvider = ({
  theme = defaultTheme,
  children,
}: ThemeProviderProps) => {
  return <SCThemeProvider theme={theme}>{children}</SCThemeProvider>;
};

export default ThemeProvider;
