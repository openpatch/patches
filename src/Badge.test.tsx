import React from "react";
import { render } from "@testing-library/react";
import { Badge, BadgeProps } from "./Badge";
import { ThemeProvider } from "./ThemeProvider";

const renderWithTheme = (props?: BadgeProps) =>
  render(
    <ThemeProvider>
      <Badge {...props} />
    </ThemeProvider>
  );

test("should render", () => {
  renderWithTheme();
});

test("should match snapshot", () => {
  const { container } = renderWithTheme();
  expect(container).toMatchSnapshot();
});
