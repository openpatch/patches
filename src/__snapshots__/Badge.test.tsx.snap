// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
.emotion-0 {
  box-sizing: border-box;
  min-width: 0;
  -webkit-transition: all 0.2s ease;
  transition: all 0.2s ease;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  font-size: 12px;
  font-weight: 600;
  padding-left: 8px;
  padding-right: 8px;
  padding-top: 4px;
  padding-bottom: 4px;
  border-radius: 9999px;
  border-style: solid;
  border-width: 1px;
  color: #0C6B58;
  background-color: #C6F7E2;
  border-color: #0C6B58;
}

<div>
  <div
    id="patches-root"
  >
    <div
      class="emotion-0"
      display="inline-flex"
      font-size="xsmall"
      font-weight="semibold"
    />
  </div>
</div>
`;
